ons_alpha/private_media/middleware.py

```diff
@@ -0,0 +1,36 @@
+from django.shortcuts import redirect
+from wagtail.admin import messages
+
+
+class PreventRootCollectionChangesMiddleware:
+    """
+    A custom Middleware that prevents editing or deletion of the special 'Private' and
+    'Public' collections used by this project to restrict access to private media.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
```

Comment:
Suggestion: Raise `MiddlewareNotUsed` when in `settings.IS_EXTERNAL_ENV`. This will reduce the extra overhead of a migration.

Reply:
@RealOrangeOne I'm not so sure about this one. The data migration creates the collections regardless of environment setup (including in tests), so I'd rather they be protected regardless too. There is no hard requirement for this item being included in `MIDDLEWARE`,  so I feel it might just be clearer to leave it out when not required. Does that make sense to you?